qase-pytest/src/qase/pytest/plugin.py

```diff
@@ -86,11 +86,12 @@ def pytest_sessionstart(self, session):
 
     def pytest_sessionfinish(self, session, exitstatus):
         if is_xdist_controller(session):
-            self.reporter.complete_run(exitstatus)
             QasePytestPlugin.drop_run_id()
         else:
             self.reporter.complete_worker()
 
+        self.reporter.complete_run(exitstatus)
```

Comment:
xdist_controller is the thread that is spawning and managing other threads, right? If we don't check for it, won't we complete the run multiple times, from each thread?

Reply:
And what about the case when the configuration is set to run.complete=False?

Reply:
No, this method is called only once regardless of threads.
The `run.complete` variable is checked further in the code.